frontend/app/routes/$lang+/_public+/apply+/$id+/child-information.tsx

```diff
@@ -0,0 +1,376 @@
+  return data ? getTitleMetaTags(data.meta.title) : [];
+});
+
+export async function loader({ context: { session }, params, request }: LoaderFunctionArgs) {
+  const state = loadApplyAdultState({ params, request, session });
+  const t = await getFixedT(request, handle.i18nNamespaces);
+
+  const csrfToken = String(session.get('csrfToken'));
+  const meta = { title: t('gcweb:meta.title.template', { title: t('apply:child-information.page-title') }) };
+
+  return json({ id: state.id, csrfToken, meta, defaultState: state.adultState.childInformation, editMode: state.adultState.editMode });
+}
+
+export async function action({ context: { session }, params, request }: ActionFunctionArgs) {
+  const log = getLogger('apply/child-information');
+
+  const state = loadApplyAdultState({ params, request, session });
+  const t = await getFixedT(request, handle.i18nNamespaces);
+
+  const formData = await request.formData();
+  const expectedCsrfToken = String(session.get('csrfToken'));
+  const submittedCsrfToken = String(formData.get('_csrf'));
+
+  if (expectedCsrfToken !== submittedCsrfToken) {
+    log.warn('Invalid CSRF token detected; expected: [%s], submitted: [%s]', expectedCsrfToken, submittedCsrfToken);
+    throw new Response('Invalid CSRF token', { status: 400 });
+  }
+
+  const formAction = z.nativeEnum(FormAction).parse(formData.get('_action'));
+
+  if (formAction === FormAction.Cancel) {
+    invariant(state.adultState.childInformation, 'Expected state.applicantInformation to be defined');
+
+    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult/review-information', params));
+  }
+
+  // Form action Continue & Save
+  // state validation schema
+  const childInformationSchema = z
+    .object({
+      firstName: z.string().trim().min(1, t('apply:child-information.error-message.first-name-required')).max(100),
+      lastName: z.string().trim().min(1, t('apply:child-information.error-message.last-name-required')).max(100),
+      dateOfBirthYear: z
+        .number({
+          required_error: t('apply:child-information.error-message.date-of-birth-year-required'),
+          invalid_type_error: t('apply:child-information.error-message.date-of-birth-year-number'),
+        })
+        .int()
+        .positive(),
+      dateOfBirthMonth: z
+        .number({
+          required_error: t('apply:child-information.error-message.date-of-birth-month-required'),
+        })
+        .int()
+        .positive(),
+      dateOfBirthDay: z
+        .number({
+          required_error: t('apply:child-information.error-message.date-of-birth-day-required'),
+          invalid_type_error: t('apply:child-information.error-message.date-of-birth-day-number'),
+        })
+        .int()
+        .positive(),
+      dateOfBirth: z.string(),
+      hasSocialInsuranceNumber: z.string().trim().min(1),
+      socialInsuranceNumber: z.string().trim().optional(),
+      isParent: z.string().trim().min(1),
```

Comment:
Should `hasSocialInsuranceNumber` and `isParent` have an error message attached in this validation schema?

Reply:
Yes, although what those are is yet to be determined.  Added in some translations.